src/components/dashboard/Receive.js

```diff
@@ -1,10 +1,151 @@
-import React from 'react'
-import { View, Text } from 'react-native'
-
-export default props => {
-  return (
-    <View>
-      <Text>Receive</Text>
-    </View>
-  )
+// @flow
+import React, { useCallback, useMemo } from 'react'
+import { Clipboard, StyleSheet, Text, View } from 'react-native'
+import { normalize } from 'react-native-elements'
+import QRCode from 'qrcode.react'
+
+import goodWallet from '../../lib/wallet/GoodWallet'
+import logger from '../../lib/logger/pino-logger'
+import { generateCode } from '../../lib/share'
+import { Address, CustomButton as Button, Section, Wrapper } from '../common'
+import { fontStyle } from '../common/styles'
+import { PushButton } from '../appNavigation/stackNavigation'
+import TopBar from '../common/TopBar'
+
+export type ReceiveProps = {
+  screenProps: any,
+  navigation: any
 }
+
+const RECEIVE_TITLE = 'Receive GD'
+
+const log = logger.child({ from: RECEIVE_TITLE })
+
+const Receive = ({ screenProps, navigation }: ReceiveProps) => {
+  const { account, networkId } = goodWallet
+  const amount = navigation.getParam('amount', '')
+
+  const code = useMemo(() => generateCode(account, networkId, amount), [account, networkId, amount])
+
+  const shareAddressAndQR = useCallback(() => log.warn('share action not yet available'))
+
+  if (!amount) {
```

Comment:
Seems like having the amount makes this component  to behave completly diferent, the layout is similar but different enough to have this big if.
I would create a new screen component for qr with amount.

Reply:
Done the refactor for the _Receive Components_

- Divided Receive into `Receive` and `ReceiveAmount`
- Exported `ShareQR` button as a component
- Updated `Dashboard` with new routes
- Exported `styles` for all the _receive_ screens